05/app.js

```diff
@@ -9,12 +9,46 @@ const stats = {
 
 /* tutaj umieść swój kod */
 
+const paragraphElements = document.querySelectorAll('p')
+const anchorElements = document.querySelectorAll('a')
+
+const paragraphArray = [...paragraphElements]
+
+paragraphArray.forEach(function(item){
+    const paragraphID = item.dataset.id
+    stats.paragraphs[paragraphID] = 0
+})
+
+const linkClickCounter = function(link){
+    const linkElement = link.getAttribute('href')
+    if(linkElement === '/dolor.html'){
+        link.addEventListener('click', function(e){
+            stats.links["/dolor.html"]++
+            
+            paragraphArray.forEach(function(item){
+                item.removeEventListener('click', paragraphClickCounter)
+            })
+            e.preventDefault();
+        })
+    }
+}
+
+const paragraphClickCounter = function(paragraph){
+    paragraph.addEventListener('click', function(){
+        const id = paragraph.dataset.id
```

Comment:
Aby nie było problemu to przed dodaniem kolejnej wartości trzeba sprawdzić czy taki "klucz" istnieje.
Jeśli nie to go tworzymy tj.
```
if(typeof stats.paragraphs[id] === 'undefined') {
    stats.paragraphs[id] = 0;
}
```